### PR TITLE
feat: use dashcore-binaries.thepasta.org as keybase.pub is defunct and validate all .asc files

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
keybase.pub isn't a thing anymore; instead use thepasta.org; also validate all .asc files

## How Has This Been Tested?
Validated 20.0.4, 20.0.3 and 20.0.2 with the script

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: contrib/verifybinaries/verify.sh [hunk 1]
```diff
@@ -25,7 +25,7 @@ TMPFILE="hashes.tmp"
 SIGNATUREFILENAME="SHA256SUMS.asc"
 RCSUBDIR="test"
 HOST1="https://github.com/dashpay/dash/releases/download/v"
-HOST2="https://pasta.keybase.pub/Dash-Core-Releases/"
+HOST2="https://dashcore-binaries.thepasta.org/file/dashcore-binaries/"
```
Comment (knst) [inline contrib/verifybinaries/verify.sh:28]: why not dash.org then?

Comment (PastaPastaPasta) [reply contrib/verifybinaries/verify.sh:28]: Dash.org doesn't host any binaries at this point at least

Comment (knst) [reply contrib/verifybinaries/verify.sh:28]: we can setup releasing process to upload this binaries before official release but not somedays later as now

Comment (PastaPastaPasta) [reply contrib/verifybinaries/verify.sh:28]: Huh? dash.org/downloads simply links to the GitHub artifacts

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has began to build this PR tagged as v20.1.0-devpr5820.f3cdbbea. A new comment will be made when the image is pushed.

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has began to build this PR tagged as v20.1.0-devpr5820.a07ba071. A new comment will be made when the image is pushed.

File: contrib/verifybinaries/verify.sh [hunk 2]
```diff
@@ -149,6 +149,26 @@ for file in $FILES
...
+          #or if a gpg error has occurred
+          echo "gpg error. Do you have the Dash Core binary release signing key installed?"
+       fi
+
+       echo "gpg output:"
+       # shellcheck disable=SC2001
+       echo "$GPGOUT"|sed 's/^/\t/g'
+       clean_up $SIGNATUREFILENAME $SIGNATUREFILENAME.2 $TMPFILE
+       exit "$RET"
+    fi
```
Comment (UdjinM6) [inline contrib/verifybinaries/verify.sh:170]:
```suggestion
   wget --quiet -N "$HOST1$BASEDIR$file.asc"

   GPGOUT=$(gpg --verify "$file.asc" 2>&1)
   RET="$?"
   if [ $RET -ne 0 ]; then
      if [ $RET -eq 1 ]; then
         #and notify the user if it's bad
         echo "Bad signature."
      elif [ $RET -eq 2 ]; then
         #or if a gpg error has occurred
         echo "gpg error. Do you have the Dash Core binary release signing key installed?"
      fi

      echo "gpg output:"
      # shellcheck disable=SC2001
      echo "$GPGOUT"|sed 's/^/\t/g'
      clean_up $SIGNATUREFILENAME $SIGNATUREFILENAME.2 $TMPFILE
      exit "$RET"
   fi
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: contrib/verifybinaries/verify.sh (hunk 2 above)
Comment (PastaPastaPasta) [reply contrib/verifybinaries/verify.sh:170]: I figured it'd be better to have "proper" syntax. The rest of the file is weirdly like 3 space tabs?

Comment (UdjinM6) [reply contrib/verifybinaries/verify.sh:170]: >The rest of the file is weirdly like 3 space tabs?

yeah, not sure why. let's either keep it 3 for consistency or convert the whole thing to 2.

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by knst — CHANGES_REQUESTED** [review]
First at all, bitcoin migrated this script from bash to python.
So, before adopt this script and writing any new code, let's migrate to python.

I backported here these changes (but not new code):
 - https://github.com/knst/dash/commit/aca17ad31b2087b8ab16ce9072848e61c9b101e4
 - https://github.com/knst/dash/commit/863625521a5a2b60c881b604593477d8a39148a9

Secondly, I got this error when I am running script from this branch:
```
$ contrib/verifybinaries/verify.sh 20.0.3
pasta.keybase.pub and github.com signature files were not equal?
```
but for version 20.0.4 - work, 20.0.2 - works, 20.0.3 - doesn't.
Is there any bug? need to fix and test it for all reasonable many version (at least  up to v19 IMO)

thirdly, Need to dashify `contrib/verifybinaries/README.md`  - that's still not dashyfied by some reason... maybe out of scope of this PR, but if you want to mention this script in any release announcement or something like that - that should be dashified before.

Fourthly, Need to dashify path for tmp binaries also: `/tmp/bitcoin_verify_binaries/` is currently - I fixed in my branch with backport (see commits)

And finally, I insist that it should be not your personal website; we have a nice candy domain dash.org and we need to utilize it!

**Comment by PastaPastaPasta** [comment]
NACK to this review

> First at all, bitcoin migrated this script from bash to python. So, before adopt this script and writing any new code, let's migrate to python.
> 
> I backported here these changes (but not new code):
> 
> * [knst@aca17ad](https://github.com/knst/dash/commit/aca17ad31b2087b8ab16ce9072848e61c9b101e4)
> * [knst@8636255](https://github.com/knst/dash/commit/863625521a5a2b60c881b604593477d8a39148a9)

I don't really think it's really valuable to migrate this to python; if you'd like to re-implement the changes from this PR as well, then I guess we can move it to python. I don't really want to reimplement my changes

> Secondly, I got this error when I am running script from this branch:
> 
> ```
> $ contrib/verifybinaries/verify.sh 20.0.3
> pasta.keybase.pub and github.com signature files were not equal?
> ```
> 
> but for version 20.0.4 - work, 20.0.2 - works, 20.0.3 - doesn't. Is there any bug? need to fix and test it for all reasonable many version (at least up to v19 IMO)

I am hosting binaries back to 18.0.0 here. I have binaries back to 0.16.0.0 on my key base that I could move over to dashcore-binaries.thepasta.org.

try running it again. Locally I've seen it fail randomly with this see
```
> $ contrib/verifybinaries/verify.sh 18.0.2
pasta.keybase.pub and github.com signature files were not equal?
                                                                                                                                      
> $ contrib/verifybinaries/verify.sh 18.0.2
stat: illegal option -- c
usage: stat [-FLnq] [-f format | -l | -r | -s | -x] [-t timefmt] [file ...]
contrib/verifybinaries/verify.sh: line 138: [: -eq: unary operator expected
Downloading dashcore-18.0.2-aarch64-linux-gnu.tar.gz
```

> thirdly, Need to dashify `contrib/verifybinaries/README.md` - that's still not dashyfied by some reason... maybe out of scope of this PR, but if you want to mention this script in any release announcement or something like that - that should be dashified before.

Won't be mentioned anywhere; just for me really

> Fourthly, Need to dashify path for tmp binaries also: `/tmp/bitcoin_verify_binaries/` is currently - I fixed in my branch with backport (see commits)

Feel free to make a new PR with this; this is a nit imo.

> And finally, I insist that it should be not your personal website; we have a nice candy domain dash.org and we need to utilize it!

Sure; but I don't control dash.org, so it would take a lot more effort to get that setup. I don't object to there being a "binaries.dash.org" but I can't really control that. Latte is looking into a potential s3 bucket or something like that that mirrors, but I see no reason why this should block. Previously it pointed to pasta.keybase.pub, now it points to dashcore-binaries.thepasta.org since keybase.pub doesn't exist. Also, this isn't REALLY meant to be for public consumption; the purpose of dashcore-binaries.thepasta.org is simply that it's a place I can store binaries but it's not taking up ~50 gigs on my local machine.

**Review by knst — COMMENTED** [review]
> Won't be mentioned anywhere; just for me really

so far as it is true, most of my comments are not relevant then and migration to dash.org is out-of-scope of this PR.

Please, check my branch that does backports and re-implement your code:

- https://github.com/knst/dash/commit/fe7cb5b583f21c2bfb5babeae191d8e300e2d1d5
- https://github.com/knst/dash/commit/13f1813a69994317d7147d93c85de38d18803434
- https://github.com/knst/dash/commit/a1c01f18da7e54308086192b3deaf2b620d87108
- https://github.com/knst/dash/commit/f7db372c448737962f5f710f2b6e52e986b02b77

----------------------------------
> I don't really think it's really valuable to migrate this to python;

There's very few people who is able to write a good code on bash and I am not one of them to be able to review it properly, yet see comments.

Beside that, the failures are difficult to debug, they have no backtrace, etc.
Just look to this random failure (from your own comment):
```
> $ contrib/verifybinaries/verify.sh 18.0.2
stat: illegal option -- c
usage: stat [-FLnq] [-f format | -l | -r | -s | -x] [-t timefmt] [file ...]
contrib/verifybinaries/verify.sh: line 138: [: -eq: unary operator expected
Downloading dashcore-18.0.2-aarch64-linux-gnu.tar.gz
```
why does it continue running and downloads `dashcore-18.0.2-aarch64-linux-gnu.tar.gz` if that's something failed?

**Comment by PastaPastaPasta** [comment]
New python version looks good!

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**